src/main/java/de/tum/cit/aet/artemis/programming/service/RepositoryAccessService.java

```diff
@@ -178,4 +179,17 @@ public void checkHasAccessToPlagiarismSubmission(ProgrammingExerciseParticipatio
         throw new AccessForbiddenException("You are not allowed to access the plagiarism result of this programming exercise.");
     }
 
+    public void checkHasAccessToOfflineIDE(ProgrammingExercise exercise, User user) throws AccessForbiddenException {
```

Comment:
Rename with suffix "ElseThrow". (Please do this to the other corresponding methods in this class as well.)